### PR TITLE
Some fixes

### PR DESCRIPTION
**Pull request by someaddons**
Closes #
Closes #
Closes #

# Changes proposed in this pull request:
- Fix racks states
- Sort schematic analyzer by total score
- Fix builder digging below himself
- Fix builder walking to position below the schematic (-1,-1,-1) rel
- Fix ToggleAble AI no longer picking up collision flags and taking long to use toggleables

[x ] Yes I tested this before submitting it.
[ ] I also did a multiplayer test.

Review please

### CODE REVIEW

File: src/main/java/com/minecolonies/api/entity/citizen/AbstractCivilianEntity.java [hunk 1]
```diff
@@ -14,6 +14,7 @@
 import net.minecraft.world.entity.player.Player;
 import net.minecraft.world.level.Level;
 import net.minecraft.world.level.block.state.BlockState;
+import net.minecraft.world.phys.Vec3;
```
Comment (Raycoms) [inline src/main/java/com/minecolonies/api/entity/citizen/AbstractCivilianEntity.java]: import

File: src/main/java/com/minecolonies/api/entity/other/AbstractFastMinecoloniesEntity.java [hunk 2]
```diff
@@ -84,6 +90,24 @@ public void setCanBeStuck(final boolean canBeStuck)
...
+     * Whether the citizen collided in the last 10 ticks
+     *
+     * @return
+     */
+    public boolean recentHorizontalCollision()
+    {
+        return level.getGameTime() - lastHorizontalCollision < 10;
+    }
+
+
```
Comment (Raycoms) [inline src/main/java/com/minecolonies/api/entity/other/AbstractFastMinecoloniesEntity.java]: double spacing

File: src/main/java/com/minecolonies/api/entity/other/AbstractFastMinecoloniesEntity.java [hunk 3]
```diff
@@ -84,6 +90,24 @@ public void setCanBeStuck(final boolean canBeStuck)
...
+        lastHorizontalCollision = level.getGameTime();
+        return super.isHorizontalCollisionMinor(vec3);
+    }
+
+    /**
+     * Whether the citizen collided in the last 10 ticks
+     *
+     * @return
+     */
+    public boolean recentHorizontalCollision()
```
Comment (Raycoms) [inline src/main/java/com/minecolonies/api/entity/other/AbstractFastMinecoloniesEntity.java]: Not sure on the name of the function =D

File: src/main/java/com/minecolonies/core/blocks/BlockMinecoloniesRack.java [hunk 4]
```diff
@@ -118,21 +119,6 @@ public VoxelShape getCollisionShape(final BlockState state, final BlockGetter le
     @Override
     public BlockState getStateForPlacement(final BlockPlaceContext context)
     {
-        final BlockPos pos = context.getClickedPos();
```
Comment (Raycoms) [inline src/main/java/com/minecolonies/core/blocks/BlockMinecoloniesRack.java:121]: getting rid of this entirely? This is how double chests are handled too

**Review by Raycoms — COMMENTED** [review]
Some comments

File: src/main/java/com/minecolonies/core/blocks/BlockMinecoloniesRack.java (hunk 4 above)
Comment (someaddons) [reply src/main/java/com/minecolonies/core/blocks/BlockMinecoloniesRack.java:121]: ye the whole state logic is handled within updateshape which is also triggered by placement

Comment (Raycoms) [reply src/main/java/com/minecolonies/core/blocks/BlockMinecoloniesRack.java:121]: works fine with schematic and buildtool too?

File: src/main/java/com/minecolonies/api/entity/other/AbstractFastMinecoloniesEntity.java (hunk 3 above)
Comment (someaddons) [reply src/main/java/com/minecolonies/api/entity/other/AbstractFastMinecoloniesEntity.java]: horizontalcollision is the name of the vanilla function it refers to, so I just added the recent

Comment (Raycoms) [reply src/main/java/com/minecolonies/api/entity/other/AbstractFastMinecoloniesEntity.java]: no I mean instead of hadRecentCollision

Comment (someaddons) [reply src/main/java/com/minecolonies/api/entity/other/AbstractFastMinecoloniesEntity.java]: well its specifically horizontal, there is also vertical

File: src/main/java/com/minecolonies/core/blocks/BlockMinecoloniesRack.java (hunk 4 above)
Comment (someaddons) [reply src/main/java/com/minecolonies/core/blocks/BlockMinecoloniesRack.java:121]: ye, thats my original testing case a schematic of a double rack with z fighting

File: src/main/java/com/minecolonies/api/entity/other/AbstractFastMinecoloniesEntity.java (hunk 3 above)
Comment (Raycoms) [reply src/main/java/com/minecolonies/api/entity/other/AbstractFastMinecoloniesEntity.java]: mean hadRecentHorizontalCollission xD

File: src/main/java/com/minecolonies/core/blocks/BlockMinecoloniesRack.java (hunk 4 above)
Comment (Raycoms) [reply src/main/java/com/minecolonies/core/blocks/BlockMinecoloniesRack.java:121]: What I had tried a lot back then was: Bunch of racks in all kinds of configurations very close proximity to eachother. Scan, rotate/mirror place a bunch of times

File: src/main/java/com/minecolonies/api/entity/other/AbstractFastMinecoloniesEntity.java (hunk 3 above)
Comment (someaddons) [reply src/main/java/com/minecolonies/api/entity/other/AbstractFastMinecoloniesEntity.java]: then probably just hadHorizontalCollission?

Comment (Raycoms) [reply src/main/java/com/minecolonies/api/entity/other/AbstractFastMinecoloniesEntity.java]: also ok

File: src/main/java/com/minecolonies/core/blocks/BlockMinecoloniesRack.java (hunk 4 above)
Comment (someaddons) [reply src/main/java/com/minecolonies/core/blocks/BlockMinecoloniesRack.java:121]: ye did the same

**Review by Raycoms — APPROVED** [review]

**Pull request merged**